src/controllers/addon_config.rs

```diff
@@ -454,6 +461,46 @@ impl FleetAddonConfig {
 
         Ok(None)
     }
+
+    async fn update_config_map(&self, ctx: Arc<Context>) -> ConfigMapSyncResult<()> {
+        let feature_gates = self
+            .spec
+            .config
+            .as_ref()
+            .map(|c| c.feature_gates.clone().unwrap_or_default())
+            .unwrap_or_default();
+
+        // If a ConfigMap reference is defined
+        if let Some(reference) = feature_gates.config_map.as_ref()
+            .and_then(|c |c.reference.as_ref()) {
+            if let Some(kind) = reference.kind.as_ref() {
+                if kind != "ConfigMap" {
+                    return Err(ConfigMapSyncError::WrongKind(kind.clone()))
+                }
+            }
+            // Get, sync, and patch the ConfigMap
+            if let (Some(name), Some(namespace)) = ((reference.name.clone()), (reference.namespace.clone())) {
+                let mut config_map = ConfigMap {
+                    metadata: ObjectMeta {
+                        name: Some(name),
+                        namespace: Some(namespace),
+                        ..Default::default()
+                    },
+                    ..Default::default()
+                }; 
+                get_or_create(ctx.clone(), &config_map).await?;
+                feature_gates.sync_configmap(&mut config_map);
+                patch(
+                    ctx,
+                    &mut config_map,
+                    &PatchParams::apply("addon-provider-fleet"),
+                )
+                .await?;
```

Comment:
```suggestion
                feature_gates.sync_configmap(&mut config_map);
                patch(
                    ctx,
                    &mut config_map,
                    &PatchParams::apply("addon-provider-fleet"),
                )
                .await?;
```
SSA patch alone should do the job.